File: lesson_22_java/src/main/java/com/lesson22java/services/CustomerService.java
package com.lesson22java.services;

import com.lesson22java.entities.Customer;
import com.lesson22java.repositories.CustomerRepository;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@AllArgsConstructor
public class CustomerService {
    private CustomerRepository customerRepository;

    public Customer create(Customer customer) {
        return customerRepository.save(customer);
    }

    public Iterable<Customer> getAll() {
        return customerRepository.findAll();
    }

    public Customer getOne(Long id) {
        return customerRepository.findById(id).orElseThrow(() -> new RuntimeException("No such customer"));
    }

    public Customer edit(Customer customer) {
        return customerRepository.save(customer);
    }

    public void delete(Long id) {
        customerRepository.deleteById(id);
    }
}

File: lesson_22/src/main/kotlin/com/lesson22/services/CustomerService.kt
package com.lesson22.services

import com.lesson22.entities.Customer
import com.lesson22.entities.Product
import com.lesson22.repositories.CustomerRepository
import org.springframework.stereotype.Service

@Service
class CustomerService (val customerRepository: CustomerRepository){
    fun getAll() = customerRepository.findAll()
    fun getById(id: Long) = customerRepository.findById(id)
    fun create(customer: Customer) = customerRepository.save(customer)
    fun delete(id: Long) = customerRepository.deleteById(id)
    fun edit(customer: Customer) = customerRepository.save(customer)
}